indexer/jobs/source_job/pg_source_job.py

```diff
@@ -0,0 +1,382 @@
+import copy
+import inspect
+import logging
+from collections import defaultdict
+from datetime import datetime
+from decimal import Decimal
+from queue import Queue
+from typing import List, Type, Union, get_args, get_origin
+
+from sqlalchemy import and_, func, or_, select
+
+from common.converter.pg_converter import domain_model_mapping
+from common.models.logs import Logs
+from common.models.transactions import Transactions
+from common.services.postgresql_service import PostgreSQLService
+from common.utils.exception_control import FastShutdownError
+from indexer.domain import Domain, dict_to_dataclass
+from indexer.domain.block import Block
+from indexer.domain.log import Log
+from indexer.domain.receipt import Receipt
+from indexer.domain.transaction import Transaction
+from indexer.jobs.base_job import BaseSourceJob
+from indexer.specification.specification import (
+    AlwaysFalseSpecification,
+    AlwaysTrueSpecification,
+    FromAddressSpecification,
+    ToAddressSpecification,
+    TransactionFilterByLogs,
+    TransactionFilterByTransactionInfo,
+    TransactionHashSpecification,
+)
+from indexer.utils.utils import flatten
+
+logger = logging.getLogger(__name__)
+
+
+class PGSourceJob(BaseSourceJob):
+    output_types = [
+        Block,
+        Transaction,
+        Log,
+    ]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._source_path = kwargs["config"].get("source_path", None)
+        self._service = PostgreSQLService(self._source_path) if self._source_path else None
+        self.pre_build = defaultdict(list)
+        self.post_build = defaultdict()
+        self.domain_mapping = defaultdict(dict)
+        self.pg_datas = defaultdict(list)
+        self._filters = flatten(kwargs.get("filters", []))
+        self._is_filter = kwargs.get("is_filter", False)
+        self._specification = AlwaysFalseSpecification() if self._is_filter else AlwaysTrueSpecification()
+
+        if self._service is None:
+            raise FastShutdownError("-pg or --postgres-url is required to run PGSourceJob")
+
+        self.build_dependency = {}
```

Comment:
What does this build_dependency do?

Reply:
This is legacy code in development and will be removed in the next commit.